index/src/index.rs

```diff
@@ -1,36 +1,52 @@
 use crate::LittIndexError::{
-    CreationError, OpenError, PdfNotFoundError, PdfParseError, WriteError,
+    CreationError, OpenError, PdfNotFoundError, PdfParseError, ReloadError, WriteError,
 };
 use crate::Result;
 use litt_shared::search_schema::SearchSchema;
 use lopdf::Document as PdfDocument;
+use std::fs::create_dir_all;
+use std::path::{Path, PathBuf};
+use tantivy::query::QueryParser;
 use tantivy::schema::{Document as TantivyDocument, Schema};
-use tantivy::{Index as TantivyIndex, IndexWriter};
+use tantivy::{Index as TantivyIndex, IndexReader, IndexWriter, ReloadPolicy, Searcher};
 use walkdir::{DirEntry, WalkDir};
 
+const INDEX_DIRECTORY_NAME: &str = ".litt-index";
+
 pub struct Index {
-    path: String,
+    documents_path: PathBuf,
     index: TantivyIndex,
+    reader: IndexReader,
     schema: SearchSchema,
 }
 
 impl Index {
-    pub fn create(path: &str, schema: SearchSchema) -> Result<Self> {
-        let index = Self::create_index(path.to_string(), schema.schema.clone())?;
+    pub fn create(path: impl AsRef<Path>, schema: SearchSchema) -> Result<Self> {
+        let documents_path = PathBuf::from(path.as_ref());
+        let index_path = documents_path.join(INDEX_DIRECTORY_NAME);
+        create_dir_all(&index_path).map_err(|e| CreationError(e.to_string()))?;
+        let index = Self::create_index(&index_path, schema.schema.clone())?;
+        let reader = Self::build_reader(&index)?;
         Ok(Self {
-            path: path.to_string(),
+            documents_path,
             index,
+            reader,
             schema,
         })
     }
 
-    pub fn open_or_create(path: &str, schema: SearchSchema) -> Result<Self> {
-        let index = Self::create_index(path.to_string(), schema.schema.clone())
-            .unwrap_or(Self::open_index(path.to_string())?);
+    pub fn open_or_create(path: impl AsRef<Path>, schema: SearchSchema) -> Result<Self> {
+        let documents_path = PathBuf::from(path.as_ref());
+        let index_path = documents_path.join(INDEX_DIRECTORY_NAME);
+        create_dir_all(&index_path).map_err(|e| CreationError(e.to_string()))?;
```

Comment:
If I understand `create_dir_all` correctly then it is only needed, " If a parent of the given path doesn’t exist" (see: https://doc.rust-lang.org/std/fs/fn.create_dir.html) 
Since the existence of the path to the index should always exist and you are only adding `.litt/` directory then you could simply use `create_dir`. 

However, if you agree that we should implement the above suggestion `.litt/index/` then `create_dir_all` is useful. (However, although the  help-page of `create_dir` indicates that `create_dir_all` is use in exactly this scenario the help-page of `create_dir_all` is at least unclear: https://doc.rust-lang.org/std/fs/fn.create_dir.html)

Reply:
Another difference is that `create_dir_all` does not throw an error if the directory already exists